### ISSUE TITLE
Rerouting crash : Null routes

### ISSUE THREAD

**Issue by Christophe668** [mapbox/mapbox-navigation-android#5338]
**Android API:** 30
**Mapbox Navigation SDK version:** 2.1.1

The SDK is crashing when rerouting using the cycling profile.

It may be related to those issues as well: 
https://github.com/mapbox/mapbox-navigation-android/issues/5051#issuecomment-956165684
https://github.com/mapbox/mapbox-navigation-android/issues/5066

I tried multiple things link the comment [here](https://github.com/mapbox/mapbox-navigation-android/issues/5051#issuecomment-956165684)
And later removing the `DirectionsCriteria.ANNOTATION_CLOSURE` as asked in the RouteOptions.

Unfortunately, I never manage to make the rerouting not crash

### Steps to trigger behavior

 Route options: 

```
		RouteOptions.builder()
			.applyDefaultNavigationOptions()
			.applyLanguageAndVoiceUnitOptions(ctx)
			.coordinates(start.toPoint(), null, end.toPoint())
			.profile(DirectionsCriteria.PROFILE_CYCLING)
			.overview(DirectionsCriteria.OVERVIEW_FALSE)
			.build()
```

### Actual behavior

`2022-01-07 15:26:11.222 4850-2099/debug I/nav-native: Directions Service: Get route request: https://api.mapbox.com/directions/v5/mapbox/cycling/4.3919126,50.8365493;4.379351,50.834826?access_token=****27CA&geometries=polyline6&overview=false&steps=true&bearings=242.8976898,90;&continue_straight=true&annotations=congestion_numeric,maxspeed,speed,duration,distance,closure&language=en&roundabout_exits=true&voice_instructions=true&banner_instructions=true&voice_units=imperial&enable_refresh=true
`

```

2022-01-07 15:26:11.446 4850-4850/bike.cowboy.app.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: bike.cowboy.app.debug, PID: 4850
    java.lang.NullPointerException: Null routes
        at com.mapbox.api.directions.v5.models.$AutoValue_DirectionsResponse.<init>($AutoValue_DirectionsResponse.java:34)
        at com.mapbox.api.directions.v5.models.AutoValue_DirectionsResponse.<init>(AutoValue_DirectionsResponse.java:25)
        at com.mapbox.api.directions.v5.models.AutoValue_DirectionsResponse$GsonTypeAdapter.read(AutoValue_DirectionsResponse.java:182)
        at com.mapbox.api.directions.v5.models.AutoValue_DirectionsResponse$GsonTypeAdapter.read(AutoValue_DirectionsResponse.java:28)
        at com.google.gson.Gson.fromJson(Gson.java:932)
        at com.google.gson.Gson.fromJson(Gson.java:897)
        at com.google.gson.Gson.fromJson(Gson.java:846)
        at com.google.gson.Gson.fromJson(Gson.java:817)
        at com.mapbox.api.directions.v5.models.DirectionsResponse.fromJson(DirectionsResponse.java:164)
        at com.mapbox.navigation.base.internal.utils.DirectionsResponseUtilsKt$parseDirectionsResponse$2.invokeSuspend(DirectionsResponseUtils.kt:29)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
```

Thanks for your time and help,

Christophe

**Comment by LukasPaczos** [comment]
Tagging @RingerJK as this might be related or already resolved by your recent default options update.

**Comment by Christophe668** [comment]
Hi @LukasPaczos. Actually I have the crash with the default options, I removed any overridden options

**Comment by cafesilencio** [comment]
Change `applyDefaultNavigationOptions()` to `applyDefaultNavigationOptions(DirectionsCriteria.PROFILE_CYCLING)` and remove `.profile(DirectionsCriteria.PROFILE_CYCLING)`

If no argument is passed to `applyDefaultNavigationOptions()` it defaults to traffic which includes `congestion_numeric` in the annotations which is an invalid annotation for cycling.  For cycling it should be `congestion`.

**Comment by RingerJK** [comment]
sorry, missed it. @cafesilencio suggestion will be available from SDK `v2.2.0` (release gonna be today), you may try with [rc](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v2.2.0-rc.1)

**Comment by cafesilencio** [comment]
Closing this as resolved.
